packages/contracts-bedrock/src/dispute/AnchorStateRegistry.sol

```diff
@@ -23,105 +24,227 @@ import { ISuperchainConfig } from "interfaces/L1/ISuperchainConfig.sol";
+        (IDisputeGame _factoryRegisteredGame,) =
+            disputeGameFactory.games({ _gameType: gameType, _rootClaim: rootClaim, _extraData: extraData });
 
-        // Must be a valid game.
-        if (address(factoryRegisteredGame) != address(game)) revert UnregisteredGame();
+        // Return whether the game is factory registered.
+        return address(_factoryRegisteredGame) == address(_game);
+    }
 
-        // No need to update anything if the anchor state is already newer.
-        if (game.l2BlockNumber() <= anchors[gameType].l2BlockNumber) {
-            return;
+    /// @notice Determines whether a game is of a respected game type.
+    /// @param _game The game to check.
+    /// @return Whether the game is of a respected game type.
+    function isGameRespected(IDisputeGame _game) public view returns (bool) {
+        return _game.wasRespectedGameTypeWhenCreated();
+    }
+
+    /// @notice Determines whether a game is blacklisted.
+    /// @param _game The game to check.
+    /// @return Whether the game is blacklisted.
+    function isGameBlacklisted(IDisputeGame _game) public view returns (bool) {
+        return portal.disputeGameBlacklist(_game);
+    }
+
+    /// @notice Determines whether a game is retired.
+    /// @param _game The game to check.
+    /// @return Whether the game is retired.
+    function isGameRetired(IDisputeGame _game) public view returns (bool) {
+        return _game.createdAt().raw() < portal.respectedGameTypeUpdatedAt();
+    }
+
+    /// @notice Returns whether a game is resolved.
+    /// @param _game The game to check.
+    /// @return Whether the game is resolved.
+    function isGameResolved(IDisputeGame _game) public view returns (bool) {
+        return _game.resolvedAt().raw() != 0
+            && (_game.status() == GameStatus.DEFENDER_WINS || _game.status() == GameStatus.CHALLENGER_WINS);
+    }
+
+    /// @notice Returns whether a game is beyond the airgap period.
+    /// @param _game The game to check.
+    /// @return Whether the game is beyond the airgap period.
+    function isGameBeyondAirgap(IDisputeGame _game) public view returns (bool) {
+        return block.timestamp - _game.resolvedAt().raw() > portal.disputeGameFinalityDelaySeconds();
+    }
+
+    /// @notice Determines whether a game resolved properly and the game was not subject to any
+    ///         invalidation conditions. The root claim of a proper game IS NOT guaranteed to be
+    ///         valid. The root claim of a proper game CAN BE incorrect and still be a proper game.
+    /// @param _game The game to check.
+    /// @return Whether the game is a proper game.
+    /// @return Reason why the game is not a proper game.
+    function isGameProper(IDisputeGame _game) public view returns (bool, string memory) {
+        if (!isGameRegistered(_game)) {
+            return (false, "game not factory registered");
```

Comment:
I recommend we use a library that defines the strings so we guarantee consistency in the API, ie clientside code will depend on these strings and we don't want to accidentally be inconsistent or change them